cdap-app-templates/cdap-etl/cdap-etl-api/src/main/java/co/cask/cdap/templates/etl/api/batch/BatchSink.java

```diff
@@ -77,4 +68,23 @@ public void configurePipeline(ETLStage stageConfig, PipelineConfigurer pipelineC
    * @param context {@link BatchSinkContext}
    */
   public abstract void prepareJob(BatchSinkContext context);
+
+  /**
+   * Initialize the sink. This is called once each time the Hadoop Job runs, before any
+   * calls to {@link #write} are made.
+   *
+   * @param stageConfig the configuration for the stage.
+   */
+  public void initialize(ETLStage stageConfig) {
```

Comment:
So BatchSinkContext here would not make sense, since it exposes hadoop job conf etc.?